#57 Add documentation for first 7 USD tutorials (#58)
Inbibo/shift_docs: docs/integration_resources/resources/usd_tutorials/usd_tutorial_04.md

<figure>
    <img src="images/usd_tutorial_04/step_02_t04.png" alt="Added 'OpenUsdFile' node to the board.">
    <figcaption><b>Figure 2</b>: Added 'OpenUsdFile' node to the board.</figcaption>
</figure>

3. To save a `.usda` file, add a new *UsdStageSave* node to the board and rename it as *SaveAsUsdaFile*. Connect the **stage** output plug from the *OpenUsdFile* node to the **stage** input plug of the new node. Set as the value for the **filepath** input plug a valid filepath to a valid `.usda` file path, such as `D:/USD_TUTORIALS/04/Sphere.usda`. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_03_t04.png" alt="Added 'SaveAsUsdaFile' node to the board connected to 'OpenUsdFile' node.">
    <figcaption><b>Figure 3</b>: Added 'SaveAsUsdaFile' node to the board connected to 'OpenUsdFile' node.</figcaption>
</figure>

> 4. Execute the workflow. Doing this, a new file with the `.usda` extension will be generated using as information the `.usd` file. The board should look like the following:

Reviewer: ```suggestion
4. Execute the workflow, which will generate a new file with the `.usda` extension using the information from the `.usd` file. The board should look like the following:
```

Author: Done

</figure>

6. To save a `.usdc` file, add a new *UsdStageSave* node to the board and rename it as *SaveAsUsdcFile*. Connect the **stage** output plug from the *OpenUsdaFile* node to the **stage** input plug of the new node. Set as the value for the **filepath** input plug a valid filepath to a `.usdc` file, such as `D:/USD_TUTORIALS/04/Sphere.usdc`. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_06_t04.png" alt="Added 'SaveAsUsdcFile' node to the board connected to 'OpenUsdaFile' node.">
    <figcaption><b>Figure 6</b>: Added 'SaveAsUsdcFile' node to the board connected to 'OpenUsdaFile' node.</figcaption>
</figure>

7. Execute the workflow again. Now, the workflow will generate a new file with the `.usdc` extension will be generated using the information from the `.usda` file. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_07_t04.png" alt="Executed the workflow for creating the '.usdc' file.">
    <figcaption><b>Figure 7</b>: Executed the workflow for creating the '.usdc' file.</figcaption>
</figure>

### Run verifications

> 1. For running the verification between the original `Sphere.usd` file and the generated file `Sphere.usda`, use the terminal for executing the *usddiff* tool shipped with USD for showing the difference between 2 USD files. The terminal should show no differences:

Reviewer: ```suggestion
1. To run the verification and inspect the differences between the original `Sphere.usd` file and the generated file `Sphere.usda`, use the terminal and execute the *usddiff* tool shipped with USD . The terminal should show no differences:
```

Author: Done

<figure>
    <img src="images/usd_tutorial_04/verification_01_t04.png" alt="Usddiff command executed between original 'Sphere.usd' file and generated 'Sphere.usda' file." width="475" height="133">
    <figcaption><b>Figure 8</b>: Usddiff command executed between original 'Sphere.usd' file and generated 'Sphere.usda' file.</figcaption>
</figure>

> 2. As previously done, for running the verification between the `Sphere.usda` file and the `Sphere.usdc` file, use the terminal for executing the *usddiff* tool. The terminal should show no differences:

Reviewer: ```suggestion
2. As previously done, to run a verification between the `Sphere.usda` file and  `Sphere.usdc`, execute the *usddiff* tool in a terminal. The terminal should show no differences:
```

Author: Done

## Converting Between Layer Formats

> This tutorial is focused on converting layer files between the different native USD file formats using the operators from the Shift_USD catalog.

Reviewer: ```suggestion
This tutorial focuses on converting layer files between the different native USD file formats using the operators from the Shift_USD catalog.
```

The base file for the example, `Sphere.usd`, can be downloaded from the [OpenUSD](https://github.com/PixarAnimationStudios/OpenUSD) Github project repository, from `extras/usd/tutorials/convertingLayerFormats/` folder.

### Converting between .usda/.usdc and .usd

1. Open Shift application. A new window's instance of Shift should appear like the following one:

<figure>
    <img src="images/usd_tutorial_04/step_01_t04.png" alt="Shift UI.">
    <figcaption><b>Figure 1</b>: Shift UI.</figcaption>
</figure>

2. First, add a new **<tt>UsdStageOpen</tt>** node to the board and rename it as `OpenUsdFile`. Using the **Inspector** widget, set as the value for the **filepath** input plug a filepath pointing to a `.usd` file. For the current example, the filepath can be: `D:/USD_TUTORIALS/04/Sphere.usd`. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_02_t04.png" alt="Added 'OpenUsdFile' node to the board.">
    <figcaption><b>Figure 2</b>: Added 'OpenUsdFile' node to the board.</figcaption>
</figure>

> 3. To save a `.usda` file, add a new **<tt>UsdStageSave</tt>** node to the board and rename it as `SaveAsUsdaFile`. Connect the **stage** output plug from the `OpenUsdFile` node to the **stage** input plug of the new node. Set as the value for the **filepath** input plug a valid filepath to a `.usda` file, for example, `D:/USD_TUTORIALS/04/Sphere.usda`. The board should look like the following:

Reviewer: Can you check the file for similar sentences? The "for example" sounds a bit unnatural in that sentence structure. "Such as" makes the sentence flow better, I don't know how to explain it haha.

```suggestion
3. To save a `.usda` file, add a new **<tt>UsdStageSave</tt>** node to the board and rename it as `SaveAsUsdaFile`. Connect the **stage** output plug from the `OpenUsdFile` node to the **stage** input plug of the new node. Set the filepath input plug to a valid .usda file path, such as D:/USD_TUTORIALS/04/Sphere.usda. The board should look like the following:
```

<figure>
    <img src="images/usd_tutorial_04/step_05_t04.png" alt="Added 'OpenUsdaFile' node to the board.">
    <figcaption><b>Figure 5</b>: Added 'OpenUsdaFile' node to the board.</figcaption>
</figure>

6. To save a `.usdc` file, add a new **<tt>UsdStageSave</tt>** node to the board and rename it as `SaveAsUsdcFile`. Connect the **stage** output plug from the `OpenUsdaFile` node to the **stage** input plug of the new node. Set as the value for the **filepath** input plug a valid filepath to a `.usdc` file, for example, `D:/USD_TUTORIALS/04/Sphere.usdc`. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_06_t04.png" alt="Added 'SaveAsUsdcFile' node to the board connected to 'OpenUsdaFile' node.">
    <figcaption><b>Figure 6</b>: Added 'SaveAsUsdcFile' node to the board connected to 'OpenUsdaFile' node.</figcaption>
</figure>

> 7. As previously done, execute the workflow again. Doing this, a new file with the `.usdc` extension will be generated using as information the `.usda` file. The board should look like the following:

Reviewer: ```suggestion
7. Execute the workflow again. Now, the workflow will generate a new file with the `.usdc` extension using the information from the `.usda` file. The board should look like the following:
```

## Converting Between Layer Formats

This tutorial focuses on converting layer files between the different native USD file formats using the operators from the Shift_USD catalog.

The base file for the example, `Sphere.usd`, can be downloaded from the [OpenUSD](https://github.com/PixarAnimationStudios/OpenUSD) Github project repository, from `extras/usd/tutorials/convertingLayerFormats/` folder.

### Converting between .usda/.usdc and .usd

1. Open Shift application. A new window's instance of Shift should appear like the following one:

<figure>
    <img src="images/usd_tutorial_04/step_01_t04.png" alt="Shift UI.">
    <figcaption><b>Figure 1</b>: Shift UI.</figcaption>
</figure>

>[!NOTE]
> Before creating the USD workflow, ensure that the USD nodes appear in the Node List widget. The catalogs combobox should show `sUSD` and `sUSDTypes` if USD is correctly loaded in Shift. If this is not the case, please check again the [Shift_USD installation page](../usd#installation).

> 2. First, add a new *UsdStageOpen* node to the board and rename it as *OpenUsdFile*. Using the **Inspector** widget, set as the value for the **filepath** input plug a filepath pointing to a `.usd` file. For the purpose of this tutorial, the filepath is: `D:/USD_TUTORIALS/04/Sphere.usd`. The board should look like the following:

Reviewer 2: ```suggestion
2. First, add a new *UsdStageOpen* node to the board and rename it as *OpenUsdFile*. Using the **Inspector** widget, set as the value for the **filepath** input plug a filepath pointing to a `.usd` file. For the purpose of this tutorial, the filepath will be: `D:/USD_TUTORIALS/04/Sphere.usd`. The board should look like the following:
```

</figure>

6. To save a `.usdc` file, add a new *UsdStageSave* node to the board and rename it as *SaveAsUsdcFile*. Connect the **stage** output plug from the *OpenUsdaFile* node to the **stage** input plug of the new node. Set as the value for the **filepath** input plug a valid filepath to a `.usdc` file, such as `D:/USD_TUTORIALS/04/Sphere.usdc`. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_06_t04.png" alt="Added 'SaveAsUsdcFile' node to the board connected to 'OpenUsdaFile' node.">
    <figcaption><b>Figure 6</b>: Added 'SaveAsUsdcFile' node to the board connected to 'OpenUsdaFile' node.</figcaption>
</figure>

7. Execute the workflow again. Now, the workflow will generate a new file with the `.usdc` extension will be generated using the information from the `.usda` file. The board should look like the following:

<figure>
    <img src="images/usd_tutorial_04/step_07_t04.png" alt="Executed the workflow for creating the '.usdc' file.">
    <figcaption><b>Figure 7</b>: Executed the workflow for creating the '.usdc' file.</figcaption>
</figure>

### Run verifications

> 1. For running the verification between the original `Sphere.usd` file and the generated file `Sphere.usda`, use the terminal for executing the *usddiff* tool shipped with USD for showing the difference between 2 USD files. The terminal should show no differences:

Reviewer 2: This will be a problem for the average user as the usd toolset shipped with shift is not sourced in any terminal. Maybe we should have a shift_console.bat ??? @Inbibo/shift
